pkg/merkle/tree_test.go

```diff
@@ -0,0 +1,331 @@
+package merkle
```

Comment:
Nit: I prefer using `merkle_test` so you can't muck with the internals of the package. There is no right or wrong way, we have just been moving a lot of tests from the the same package into its own (_test) for exactly this reason.